sources/api/thar-be-updates/src/main.rs

```diff
@@ -0,0 +1,562 @@
+    status.set_recent_command_info(UpdateCommand::PrepareUpdate, Some(&output), None);
+    if output.status.success() {
+        status.set_staging_partition_image_info(chosen_update);
+    } else {
+        warn!("Failed to prepare the update with updog");
+    }
+    Ok(())
+}
+
+/// "Activates" the staged update by setting the appropriate priority flags in the partition table
+fn activate_update(status: &mut UpdateStatus<update_metadata::Update>) -> Result<()> {
+    let mut gpt_state = State::load().context(error::PartitionTableRead)?;
+    gpt_state
+        .upgrade_to_inactive()
+        .context(error::InactivePartitionUpgrade)?;
+    gpt_state.write().context(error::PartitionTableWrite)?;
+    status.mark_staging_partition_next_to_boot()?;
+    Ok(())
+}
+
+/// "Deactivates" the staged update by rolling back actions done by `activate_update`
+fn deactivate_update(status: &mut UpdateStatus<update_metadata::Update>) -> Result<()> {
+    let mut gpt_state = State::load().context(error::PartitionTableRead)?;
+    // This actually wipes all the priority bits in the inactive partition
+    gpt_state.cancel_upgrade();
+    // The update image is still there in the inactive partition so mark as valid
+    gpt_state.mark_inactive_valid();
+    gpt_state.write().context(error::PartitionTableWrite)?;
+    status.unmark_staging_partition_next_to_boot()?;
+    Ok(())
+}
+
+/// Gets the update status, serialize the structure to json and output to stdout.
+fn report_update_status(lockfile: File) -> Result<()> {
+    lock_shared(&lockfile)?;
+    let update_status = get_update_status()?;
+    unlock(&lockfile)?;
+    println!(
+        "{}",
+        // We simplify the available_updates information into just the version numbers when reporting status
+        serde_json::to_string_pretty(&update_status.simplify_update_info())
+            .context(error::StatusSerialize)?
+    );
+    Ok(())
+}
+
+/// Reports all the available updates along with their detailed information
+fn report_available_updates(lockfile: File) -> Result<()> {
+    lock_shared(&lockfile)?;
+    let update_status = get_update_status()?;
+    unlock(&lockfile)?;
+    println!(
+        "{}",
+        // We simplify the available_updates information into just the version numbers when reporting status
+        serde_json::to_string_pretty(&update_status.get_available_updates())
+            .context(error::StatusSerialize)?
+    );
+    Ok(())
+}
+
+/// Reports detailed information about a specific update version
+fn report_detailed_update_info(lockfile: File, version: FriendlyVersion) -> Result<()> {
+    lock_shared(&lockfile)?;
+    let update_status = get_update_status()?;
+    unlock(&lockfile)?;
+    let version_str: String = version.to_owned().into();
+    let update_info = if version_str == "latest" {
+        // Get the information for the 'latest' update
+        update_status
+            .get_latest_update()?
+            .context(error::QueriedUpdateVersion { version })?
```

Comment:
If we're already on the latest image version, this would error, which would be somewhat confusing. This should probably be a different error.